python/ray/experimental/data/read_api.py

```diff
@@ -211,17 +211,29 @@ def gen_read(pieces: List["pyarrow._dataset.ParquetFileFragment"]):
     metadata: List[BlockMetadata] = []
     for pieces in nonempty_tasks:
         calls.append(lambda pieces=pieces: gen_read.remote(pieces))
-        piece_metadata = [p.metadata for p in pieces]
-        metadata.append(
-            BlockMetadata(
+        input_files = [p.path for p in pieces]
+        piece_metadata = []
+        for p in pieces:
```

Comment:
Can we avoid for-else construct? I never remember how it works, and expect others will be similarly confused.